### DISCUSSION TITLE
PRs naming convention

### DISCUSSION THREAD

**Discussion by Bertaz** [easylist/easylist#16755] [Q&A] [answered]
I can see most PRs here follow a naming convention: `[A/M/P]: https://example.com (fix #0)`. E.g. [`P: https://webtrack.dhlecs.com (Fix #14794)`](https://github.com/easylist/easylist/pull/16742).
While the site part and the fixed issue part are self explanatory, I don't fully understand the starting abbreviations.
As this repo has no contributing guidelines, this (organic?) convention isn't explained anywhere, so it would be good to have it written somewhere people can reference it.

I guess A: stands for addition of a filter, M: for modification of a filter. P: not sure, added filter exception? Are there more?

**Comment by Nojuuu** [comment]
P = problem

**Reply by Bertaz** [reply]
And how would you use that? Is it when you modify or remove a problematic filter?
What about A and M, is my guess correct?
If you add an exception to an allowlist like [easylist_cookie/easylist_cookie_allowlist.txt](https://github.com/easylist/easylist/blob/67115172df259e817f6b121a8368c8776064b7d1/easylist_cookie/easylist_cookie_allowlist.txt) should I use A: (as I'm adding a new filter) or P: (as I'm adding an exception to an existing filter)?
What if you remove a filter? M:?

**Reply by Nojuuu** [reply]
I'm not maintainer here, so I don't know all of the official guidelines, but...


> And how would you use that? Is it when you modify or remove a problematic filter?

I usually use P: when there is a more noticeable breakage that prevents from using the page properly, for example when a section doesn't load, scrolling doesn't work, logging in doesn't work, etc.

The needed fix in these cases could be anything, add/modify/remove.

> What about A and M, is my guess correct?

Yes it is correct.

> If you add an exception to an allowlist like [easylist_cookie/easylist_cookie_allowlist.txt](https://github.com/easylist/easylist/blob/67115172df259e817f6b121a8368c8776064b7d1/easylist_cookie/easylist_cookie_allowlist.txt) should I use A: (as I'm adding a new filter) or P: (as I'm adding an exception to an existing filter)?

If you look at the commit history, it looks like it's usually been P: for bigger problems and sometimes for smaller issues A: or M: if you modify an existing filter or add a new domains to it.

> What if you remove a filter? M:?

M: or P: depending on the situation, M: is often used when you remove outdated filters.

**Reply by Bertaz** [reply]
Thank you for the explanation

**Comment by Yuki2718** [comment, marked as answer]
https://github.com/easylist/easylist/pull/14774#issuecomment-1412244081
